src/mobile-token/MobileTokenContext.tsx

```diff
@@ -224,10 +252,14 @@ export const MobileTokenContextProvider: React.FC = ({children}) => {
           });
         });
       } finally {
+        // We've finished with remote tokens. Cancel timeout notification.
+        cancelTimeoutHandler();
+        setTimeoutFlag(false);
+
         setIsLoading(false);
       }
     }
-  }, [enabled, loadNativeToken, loadRemoteTokens]);
+  }, [enabled, loadNativeToken, loadRemoteTokens, token_timeout_in_seconds]);
 
   useEffect(() => {
     load();
```

Comment:
not sure if we should return in this useEffect something like

```
const cancelTimeoutHandler = load();
return () => cancelTimeoutHandler();
```

Reply:
Hmm. I think that either its cancelled in the `finally` block, or it is executed in a sense. I think that in any sense (even if the useEffect block changes dependency) the current execution flow will finish and not stop "mid execution" if that makes sense